### PR TITLE
[Merged by Bors] - feat: port Order.WellFoundedSet

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/WellFoundedSet.lean [hunk 1]
```diff
@@ -0,0 +1,811 @@
...
+import Mathlib.Order.WellFounded
+import Mathlib.Data.List.TFAE
+
+/-!
+# Well-founded sets
+
+A well-founded subset of an ordered type is one on which the relation `<` is well-founded.
+
+## Main Definitions
+ * `Set.WellFoundedon s r` indicates that the relation `r` is
```
Comment (dupuisf) [inline Mathlib/Order/WellFoundedSet.lean:22]:
```suggestion
 * `Set.WellFoundedOn s r` indicates that the relation `r` is
```

File: Mathlib/Order/WellFoundedSet.lean [hunk 2]
```diff
@@ -0,0 +1,811 @@
...
+
+variable {ι α β : Type _}
+
+namespace Set
+
+/-! ### Relations well-founded on sets -/
+
+
+/-- `s.well_founded_on r` indicates that the relation `r` is well-founded when restricted to `s`. -/
+def WellFoundedOn (s : Set α) (r : α → α → Prop) : Prop :=
```
Comment (dupuisf) [inline Mathlib/Order/WellFoundedSet.lean:59]:
```suggestion
/-- `s.WellFoundedOn r` indicates that the relation `r` is well-founded when restricted to `s`. -/
def WellFoundedOn (s : Set α) (r : α → α → Prop) : Prop :=
```

File: Mathlib/Order/WellFoundedSet.lean [hunk 3]
```diff
@@ -0,0 +1,811 @@
...
+end WellFoundedOn
+
+/-! ### Sets well-founded w.r.t. the strict inequality -/
+
+section LT
+
+variable [LT α] {s t : Set α}
+
+/-- `s.is_wf` indicates that `<` is well-founded when restricted to `s`. -/
+def IsWf (s : Set α) : Prop :=
```
Comment (dupuisf) [inline Mathlib/Order/WellFoundedSet.lean:183]:
```suggestion
/-- `s.IsWf` indicates that `<` is well-founded when restricted to `s`. -/
def IsWf (s : Set α) : Prop :=
```

File: Mathlib/Order/WellFoundedSet.lean [hunk 4]
```diff
@@ -0,0 +1,811 @@
...
+end Finset
+
+namespace Set
+
+section Preorder
+
+variable [Preorder α] {s : Set α} {a : α}
+
+/-- `is_wf.min` returns a minimal element of a nonempty well-founded set. -/
+noncomputable nonrec def IsWf.min (hs : IsWf s) (hn : s.Nonempty) : α :=
```
Comment (dupuisf) [inline Mathlib/Order/WellFoundedSet.lean:615]:
```suggestion
/-- `IsWf.min` returns a minimal element of a nonempty well-founded set. -/
noncomputable nonrec def IsWf.min (hs : IsWf s) (hn : s.Nonempty) : α :=
```

**Review by dupuisf — COMMENTED** [review]
Looks good!

bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**